drivers/include/lis2dh12.h

```diff
@@ -101,8 +113,58 @@ typedef struct {
 #endif
     lis2dh12_scale_t scale;         /**< sampling sensitivity used */
     lis2dh12_rate_t rate;           /**< sampling rate used */
+    lis2dh12_powermode_t powermode; /**< power mode used*/
 } lis2dh12_params_t;
 
+/**
+ * @brief   LIS2DH12 high pass modes
+ */
+typedef enum {
+    LIS2DH12_HP_MODE_NORMAL    = 0b00, /**< normal mode, reset by reading REG_REFERENCE */
+    LIS2DH12_HP_MODE_REFERENCE = 0b01, /**< uses the reference signal for filtering */
+    LIS2DH12_HP_MODE_AUTORESET = 0b11, /**< automatically resets on interrupt generation */
+} lis2dh12_hp_mode_t;
+
+/**
+ * @brief   LIS2DH12 high pass cutoff frequency
+ */
+typedef enum {
+    LIS2DH12_HP_FREQ_DIV50  = 0, /**< cutoff freq is ODR divided by 50 */
+    LIS2DH12_HP_FREQ_DIV100 = 1, /**< cutoff freq is ODR divided by 100 */
+    LIS2DH12_HP_FREQ_DIV200 = 2, /**< cutoff freq is ODR divided by 200 */
+    LIS2DH12_HP_FREQ_DIV400 = 3, /**< cutoff freq is ODR divided by 400 */
+} lis2dh12_hp_freq_t;
+
+/**
+ * @brief   LIS2DH12 high pass config values
+ */
+typedef struct {
+    lis2dh12_hp_mode_t Highpass_mode; /**< set the High pass mode */
+    lis2dh12_hp_freq_t Highpass_freq; /**< set the High pass cutoff frequency \
+                                            related to device rate */
+    bool CLICK_enable;    /**< enables filter for click data */
+    bool INT1_enable;     /**< enables filter for AOI on interrupt 1 */
+    bool INT2_enable;     /**< enables filter for AOI on interrupt 2 */
+    bool DATA_OUT_enable; /**< enables filter for data output */
+} lis2dh12_highpass_t;
+
+/**
+ * @brief   LIS2DH12 click config values
+ */
+typedef struct {
+    bool enable_DOUBLE;     /**< otherwise single click for given axis are enabled */
+    bool enable_X_CLICK;    /**< enable double pr single click for X axes */
+    bool enable_Y_CLICK;    /**< enable double pr single click for Y axes */
+    bool enable_Z_CLICK;    /**< enable double pr single click for Z axes */
+    bool noINT_latency;     /**< if "0" interrupt stays high for TIME_latency setting \
+                                if "1" interrupt stays high until CLICK_SRC is read */
+    uint8_t curr_ODR_Hz;    /**< give current ODR setting (current frequency in Hz) */
```

Comment:
yes, the interval time is much better.

Reply:
I changed it to number of ODR cycles. Then the user should deal with it. Because if we do have to handle the interval time we should be able to do this in µs (>1000Hz). This will blow up all time values to `uint32_t` because we have to support the 1Hz (1sec) as well.